cores/esp8266/uart.c

```diff
@@ -422,15 +426,15 @@ uart_tx_free(uart_t* uart)
     return UART_TX_FIFO_SIZE - uart_tx_fifo_available(uart->uart_nr);
 }
 
-void 
+uint8_t 
 uart_wait_tx_empty(uart_t* uart)
 {
     if(uart == NULL || !uart->tx_enabled)
-        return;
-
+        return 0;
+    uint8_t bit_length = MODE2WIDTH(USC0(uart->uart_nr)) + MODE2PARITY(USC0(uart->uart_nr)) + MODE2STOP(USC0(uart->uart_nr)) + 1;
```

Comment:
This calculation has nothing to do with the function in which it is being executed. I suggest reverting uart_wait_tx_empty(), and putting this code in its own function, e.g.:
```
inline uint8_t uart_get_bit_length(const int uart_nr)
{
  return MODE2WIDTH(USC0(uart_nr)) + MODE2PARITY(USC0(uart_nr)) + MODE2STOP(USC0(uart_nr)) + 1;  //+1 for the start bit, which is always there
}
```

Then, use it in HardwareSerial like so.
```
     uart_wait_tx_empty(_uart);	
     //Workaround for a bug in serial not actually being finished yet
     //Wait for 8 data bits, 1 parity and 2 stop bits, just in case	  
     delayMicroseconds(1000000 * uart_get_bit_length(_uart->uart_nr) / uart_get_baudrate(_uart) + 1);
```